src/main/java/nl/hsac/fitnesse/symbols/RandomString.java

```diff
@@ -41,13 +41,56 @@ public RandomString() {
     }
 
     public String toTarget(Translator translator, Symbol symbol) {
+        String param = symbol.getProperty(MIN_MAX_LENGTH, null);
         String permitted = symbol.getProperty(CHARACTERS, DEFAULT_CHARS);
-        int length = getProperty(symbol, LENGTH, -1);
         String prefix = symbol.getProperty(PREFIX, "");
-        length = length - prefix.length();
-        if (length < 1) {
-            length = RANDOM_UTIL.random(100);
-        }
+        int length = getRandomStringLength(param, prefix);
         return prefix + RANDOM_UTIL.randomString(permitted, length);
     }
+
+    public int getRandomStringLength(String param, String prefix) {
+        int randomStringLength;
+        int minimalLength;
+        int maximalLength;
+        int prefixLength;
+
+        //Handle the prefix input parameter
+        if (prefix == "") {
+            prefixLength = 0;
+        } else {
+            prefixLength = prefix.length();
+        }
+
+        // Handle the length input parameter
+        if (param == null) { //if there is no parameter, use the default
+            randomStringLength = RANDOM_UTIL.random(100);
+        } else if (!param.contains(",")) { //no comma in the parameter, just parse directly to requestedLength
```

Comment:
Doing a negative check (with a `!`) is harder to understand than just a normal one. If you remove the `!` and then which the bodies of the `else if` en `else` branches the code is easier to read.